supergloo/docs/mesh/mesh-discovery.md

```diff
@@ -0,0 +1,146 @@
+---
+title: "Mesh Discovery"
+weight: 2
+---
+
+# Overview
+
+Mesh discovery is the ability to discovery service meshes which are running in the cluster to which mesh
+discovery is deployed. This capability is currently shipped with supergloo by default, but in the future
+will be available as a standalone feature.
+
+Currently supported meshes for Discovery:
+
+- Istio
+
+# Architecture
+
+Mesh discovery, similar to other solo.io projects, uses an event loop based architecure to watch kubernetes 
+resources to create/update those resources. In this case we are interested in three resources:
+
+* [Mesh](../../v1/github.com/solo-io/supergloo/api/v1/mesh.proto.sk)
+* [Install](../../v1/github.com/solo-io/supergloo/api/v1/install.proto.sk)
+* [Pod](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#pod-v1-core) (kubernetes resource)
+
+More information about these resources can be found by clicking the links above. 
+
+At its core mesh discovery monitors these resources and takes actions based on certain heuristics. These heuristics 
+vary slightly by the type of mesh we are trying to discover, but the concept is similar. For example, in the case of istio: 
+mesh discovery watches for the existence of a deployment named istio-pilot and discovers the deployed Istio version based 
+on Pilot's image tag.
```

Comment:
So let's try be consistent with the words used. Let's use "SuperGloo" to refer to SuperGloo and "Kubernetes" and Istio, etc. Some are capitalized, some are not, etc .. let's update them throughout the doc to be consistent. 

Also, solo.io should be Solo.io